### PR TITLE
WIP, BUG: exp, log AVX loops do not use steps

### LINKED ISSUE #13512: BUG: exp and log with non-contiguous float32 inputs
It seems to me that the AVX routines of `np.exp` and `np.log` disregard the `strides` attribute of inputs.

### Reproducing code example:

```python
import numpy as np
print(np.exp(np.arange(4, dtype=np.float64)[::2]))  # correct
print(np.exp(np.arange(4, dtype=np.float32)[::2]))  # wrong
print(np.exp(np.arange(4, dtype=np.float16)[::2]))  # correct
```

output:
```
[1.        7.3890561]
[1.       2.718282]
[1.   7.39]
```

### Numpy/Python version information:

```
1.17.0.dev0+634d66d 3.7.1 (default, Nov  6 2018, 21:02:07)
[Clang 10.0.0 (clang-1000.10.44.4)]
```

### PR DESCRIPTION
**Pull request by mattip**
Fixes #13512 by considering steps in the inner loop

Still a WIP - added a test for all ufuncs

### CODE REVIEW

**Comment by mattip** [comment]
Still need to work `steps` into `@ISA@_exp_FLOAT` ane `@ISA@_log_FLOAT` to make the test pass

**Comment by mattip** [comment]
@r-devulap you can issue a PR to this branch or start a new branch with some form of this test

<details>

To issue a PR against this branch (assuming I have no bugs in this code):

```
git remote add mattip https://github.com/mattip/numpy
git fetch mattip issue13512
git checkout issue12512
... (fix the bug, check benchmarks and *so size change)
git push r-devulap
```
then go to ` https://github.com/r-devulap/numpy/pull/new/issue13512`

**Comment by r-devulap** [comment]
@mattip thanks. I am working on a fix and will submit them to this PR.

File: numpy/core/tests/test_ufunc.py [hunk 1]
```diff
@@ -1964,3 +1964,28 @@ def test_ufunc_types(ufunc):
                 assert r.dtype == np.dtype(t)
         else:
             assert res.dtype == np.dtype(out)
+
+@pytest.mark.parametrize('ufunc', [getattr(np, x) for x in dir(np)
+                                if isinstance(getattr(np, x), np.ufunc)])
+def test_ufunc_noncontiguous(ufunc):
```
Comment (juliantaylor) [inline numpy/core/tests/test_ufunc.py]: a ufunc test for overlapping input and output memory is needed for vectorization too
`_gen_alignment_data` produces a bunch of data for unary and binary simd tests  (though not strided)
but it creates a huge amount of test cases which makes tests slow

nowadays atleast on x86 you don't need the alignment stuff anymore so maybe overlapping input and output for unary and binary functions is enough?

Comment (mattip) [reply numpy/core/tests/test_ufunc.py]: Would it be enough to set `func(*args, out=arg[0])` or would you like to see offset overlap as well?

**Comment by mattip** [comment]
Duplicate of #13520, closing

**Pull request closed without merging**